packages/input-datepicker/src/LionInputDatepicker.js

```diff
@@ -213,12 +216,19 @@ export class LionInputDatepicker extends LionInputDate {
     }
   }
 
+  __onReadOnlyChanged() {
+    if (this._invokerElement) {
+      this._invokerElement.disabled = this.disabled || this.readOnly;
+    }
+  }
```

Comment:
for disabled you have this

```js
  __delegateDisabled() {
    if (this.delegations.target()) {
      this.delegations.target().disabled = this.disabled;
    }
    if (this._invokerElement) {
      this._invokerElement.disabled = this.disabled;
    }
  }
```

why not do the same for readonly and also set `this.delegations.target().readOnly`?

Reply:
the invokerElement (button) is not configured as delegation target. The inputElement is

Reply:
But `delegations.target()` should be called `this.inputElement` to be forwards compatible with refactoring about DelegateMixin

Reply:
does not explain why you don't delegate 'readonly' to the input element? I didn't ask to set such an attribute on the invoker, but rather delegate and set it on the input element

by "delegate" I don't mean using DelegateMixin, I mean literally the delegation itself

let's agree on the name of this method after we agree on the logic here

Reply:
No, because delegation is `source[propX] = target[propX] `. This was `source[propX] = target[propY] `, so observer logic